tests/plugins/test_gen_design_doc_impl.py

```diff
@@ -13,7 +14,8 @@ def test_can_generate_design_doc_with_models(self):
             with open(test_model_file_name, "w") as arch_file:
                 arch_file.write(TEST_MODEL)
 
-            gen_design_doc(test_model_file_name, temp_dir)
+            result = gen_design_doc(test_model_file_name, temp_dir)
+            self.assertEqual(result.status_code, PluginExecutionStatusCode.SUCCESS)
```

Comment:
Should add plugin failure/error handling. There could be IO issues that cause the plugin to fail.